src/executables/sysrepo-plugind.c

```diff
@@ -527,6 +534,34 @@ main(int argc, char **argv)
         goto cleanup;
     }
 
+    /* open and lock pid file before plugins are loaded, but don't update it yet 
+     * plugins may fail, and do some initialization that must finish before we are
+     * considered up and running. No need to create a pid file if we're in debug
+     * mode (we're not a daemon)
+     */
+
+    if (debug && pidfile) {
+        SRPLG_LOG_WRN("sysrepo-plugind", "Ignoring pidfile in debug mode.");
+    }
+
+    if (!debug && pidfile) {
+        /* make sure we are the only instance - lock the PID file and write the PID */
```

Comment:
this comment is misleading because *this* code doesn't write anything; it only happens later on, after the plugins have been loaded